spark/src/main/java/org/apache/iceberg/spark/source/RowDataRewriter.java

```diff
@@ -106,8 +107,15 @@ public RowDataRewriter(Table table, PartitionSpec spec, boolean caseSensitive,
     if (spec.fields().isEmpty()) {
       writer = new UnpartitionedWriter<>(spec, format, appenderFactory, fileFactory, io.value(), Long.MAX_VALUE);
     } else {
-      writer = new SparkPartitionedWriter(spec, format, appenderFactory, fileFactory, io.value(), Long.MAX_VALUE,
-          schema, structType);
+      if (PropertyUtil.propertyAsBoolean(properties,
```

Comment:
nit:  I think it's more clear to just use : 

```java
if(spec.fields().isEmpty()){
   return UnpartitionedWriter
} else if(xxx){
    return SparkPartitionedFanoutWriter ; 
} else {
    return SparkPartitionedWriter
}
```

Reply:
well